MyStore/src/JsonParcer.cpp

```diff
@@ -0,0 +1,77 @@
+#include "JsonParcer.h"
+
+
+bool JsonParcer::Deserialization()
+{
+	std::ifstream file(this->m_file_path);
+
+	if (this->m_file_path.empty() || !file)
+		return false;
+
+	std::string text = {};
+	
+	while (!file.eof())
+	{
+		if (this->m_data.size() == 5000) //It has to support up to 5000 products in database.
+			return false;
+
+		getline(file, text);
+
+		if (text.find("name") != std::string::npos) 
+		{
+			for (int i = 0; i < text.size(); i++)
+			{
+				if (text[i] == ':')
+				{
+					int index = text.find(':');
+					int left_index = text.find('"', index);
+					int right_index = text.find('"', left_index + 1);
+					this->Product.m_product_name = text.substr(left_index + 1, right_index-left_index - 1);
+					//need to check amount of characters, limit is 20
+				}
+			}
+		}
+		if (text.find("color") != std::string::npos)
+		{
+			for (int i = 0; i < text.size(); i++)
+			{
+				if (text[i] == ':')
+				{
+					int index = text.find(':');
+					int left_index = text.find('"', index);
+					int right_index = text.find('"', left_index + 1);
+					this->Product.m_product_color = text.substr(left_index + 1, right_index - left_index - 1);
+
+					if (Product.m_product_color.size() > 20 || Product.m_product_name.size() > 20) 					//need to check amount of characters, limit is 20

```

Comment:
It's better to declare 20 as some constant so that if this requirement changes, you can easily update it in one place instead of searching for it across entire codebase.